src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
```

Comment:
PullProofAI

The `findDuplicates` method has a time complexity of O(n^2) due to nested loops, which can be inefficient for large arrays. Consider optimizing this method to reduce the time complexity for finding duplicates.

One approach is to use a `Set` to store unique elements and another array to store duplicates, resulting in a more efficient solution with a time complexity of O(n).

```typescript
public findDuplicates(arr: number[]): number[] {
    const set = new Set<number>();
    const duplicates: number[] = [];

    for (const num of arr) {
        if (set.has(num)) {
            duplicates.push(num);
        } else {
            set.add(num);
        }
    }

    return duplicates;
}
```

By using a `Set` to track unique elements, the method avoids the need for nested loops and improves performance significantly.



<!-- This is an auto-generated comment by PullProofAI -->